haystack/schema.py

```diff
@@ -16,21 +15,15 @@
 import time
 import json
 import ast
-from dataclasses import asdict
+from dataclasses import asdict, InitVar
 
 import mmh3
 import numpy as np
 import pandas as pd
 
-from pydantic import BaseConfig
+from pydantic import BaseConfig, Field
 from pydantic.json import pydantic_encoder
-
-if not typing.TYPE_CHECKING:
-    # We are using Pydantic dataclasses instead of vanilla Python's
-    # See #1598 for the reasons behind this choice & performance considerations
-    from pydantic.dataclasses import dataclass
-else:
-    from dataclasses import dataclass  # type: ignore  # pylint: disable=ungrouped-imports
+from pydantic.dataclasses import dataclass
```

Comment:
Let's retain the comment, it points to a useful analysis of Pydantic `dataclass` vs Python `dataclass`.

```suggestion
# We are using Pydantic dataclasses instead of vanilla Python's
# See #1598 for the reasons behind this choice & performance considerations
from pydantic.dataclasses import dataclass
```